src/test/scala/com/databricks/spark/csv/util/InferSchemaSuite.scala

```diff
@@ -52,4 +83,14 @@ class InferSchemaSuite extends FunSuite {
       Array(LongType)).deep == Array(DoubleType).deep)
   }
 
+  test("Type/Schema inference works as expected for the simple sparse dataset.")
```

Comment:
Please move this test to `CsvSuite.scala`. In that suite, there is already a test SQL context, and you don't need to construct SQLContext in this file.